### PR TITLE
Add improvements and bug fixes to Safe Control

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Compilation of fixes and improvements as a result of testing on QA

**Changes** 🏗

* Added safeguards to IPFS metadata fetching
* Added metatransaction messages for gas stations related to Safe Control
* Fixed "Unknown contract" appearing in the event list of ActionPage as "Unknown contract contract"

### CODE REVIEW

**Review by chinins — APPROVED** [review]
Very good!

**Review by willm30 — APPROVED** [review]
LGTM bro

File: src/data/resolvers/motions.ts [hunk 1]
```diff
@@ -1310,20 +1310,21 @@ export const motionsResolvers = ({
           apolloClient,
         );
 
-        const safeTxMetadata = JSON.parse(
-          (await getColonyMetadataIPFS(annotation?.values?.metadata || '')) ||
-            '[]',
-        );
+        if (annotation) {
+          const safeTxMetadata = await getColonyMetadataIPFS(
```
Comment (danbr) [inline src/data/resolvers/motions.ts:1314]: I'm a bit surprised with getColonyMetadataIPFS It doesnt really make use of ColonyMetadataEventParser.
The func is returning (promise)any, but could be (type) ColonyMetadata as the name suggests.

Also, inside getColonyMetadataIPFS it doesnt handle different version of metadataVersion. I guess in theory we would not expect any safe data to be encoded in V1, everything should now be V2, but its good to have the framework so we can easily support V3.

Not using ColonyMetadataEventParser to parse V2 data means that yup validation is not be used.

What happens if the data is malformed?

If you're using the annotation to store data, why not model the function on: `ActionsPageFeedItemWithIPFS.tsx` around line 51, which already implements much of the parsing/validation?

**Review by danbr — COMMENTED** [review]
Left a few comments, I dont know the details of the specific case, but perhaps you can make use of the ColonyEventMetadataParser?

**Comment by ArmandoGraterol** [comment]
@danbr For some reason, when I merged #4067 this branch was also merged?? Don't know what happened there, but don't worry, I will address your feedback in another PR

**Pull request merged**